src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,773 @@
+							treeViewList.SetDraggedOverItem(draggedOverItem);
+						}
+
+						treeViewList.UpdateDropTargetDropEffect(false, false, null);
+					}
+				}
+			}
+
+			base.OnDragOver(args);
+		}
+
+		protected override void OnDragEnter(DragEventArgs args)
+		{
+			TreeViewItem draggedOverItem = this;
+
+			args.AcceptedOperation = DataPackageOperation.None;
+			args.DragUIOverride.IsGlyphVisible = true;
+
+			var treeView = AncestorTreeView;
+			if (treeView != null && treeView.CanReorderItems && !args.Handled)
+			{
+				var treeViewList = treeView.ListControl;
+				TreeViewNode draggedNode = treeViewList.DraggedTreeViewNode;
+				if (draggedNode != null)
+				{
+					TreeViewNode draggedOverNode = treeView.NodeFromContainer(draggedOverItem);
+					TreeViewNode walkNode = draggedOverNode.Parent;
+
+					while (walkNode != null && walkNode != draggedNode)
+					{
+						walkNode = walkNode.Parent;
+					}
+
+					if (walkNode != draggedNode && draggedNode != draggedOverNode)
+					{
+						args.AcceptedOperation(DataPackageOperation.Move);
+					}
+
+					TreeViewNode droppedOnNode = TreeNode;
+					if (droppedOnNode != draggedNode) // Don't expand if drag hovering on itself.
+					{
+						// Set up timer.
+						if (!draggedOverNode.IsExpanded && draggedOverNode.HasChildren)
+						{
+							if (m_expandContentTimer != null)
+							{
+								var expandContentTimer = m_expandContentTimer;
+								expandContentTimer.Stop();
+								expandContentTimer.Start();
+							}
+							else
+							{
+								// Initialize timer.
+								TimeSpan interval = TimeSpan::duration(c_dragOverInterval);
+								var expandContentTimer = new DispatcherTimer();
+								m_expandContentTimer.set(expandContentTimer);
+								expandContentTimer.Interval = interval;
+								expandContentTimer.Tick += OnExpandContentTimerTick;
+								expandContentTimer.Start();
+							}
+						}
+					}
+				}
+			}
+
+			base.OnDragEnter(args);
+		}
+
+		protected override void OnDragLeave(DragEventArgs args)
+		{
+			if (!args.Handled)
+			{
+				var treeView = AncestorTreeView;
+				if (treeView != null)
+				{
+					var treeViewList = treeView.ListControl;
+					treeViewList.SetDraggedOverItem(null);
+				}
+
+				if (m_expandContentTimer)
+				{
+					m_expandContentTimer.Stop();
+				}
+			}
+			base.OnDragLeave(args);
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer() =>
+			new TreeViewItemAutomationPeer(this);
+
+		// IFrameworkElementOverrides
+
+		protected override void OnApplyTemplate()
+		{
+			RecycleEvents();
+
+			IControlProtected controlProtected = *this;
+			m_selectionBox = (CheckBox)GetTemplateChild(c_multiSelectCheckBoxName);
+			RegisterPropertyChangedCallback(winrt::SelectorItem::IsSelectedProperty(), { this, &TreeViewItem::OnIsSelectedChanged });
+
+			if (m_selectionBox)
+			{
+				m_checkedEventRevoker = m_selectionBox.get().Checked(winrt::auto_revoke, { this, &TreeViewItem::OnCheckToggle });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)